scheduler/scheduler.go

```diff
@@ -125,18 +126,18 @@ func (fq *fifoQueue) status() QueueStatus {
 	return QueueStatus{
 		ActiveRequests: int(active),
 		QueuedRequests: int(queued),
-		Closed:         fq.closed,
+		Closed:         closed,
 	}
 }
 
-// Reconfigure does nothing, because it creates a data race that would
-// require more mutex than we want to have
 func (fq *fifoQueue) reconfigure(c Config) {
 	fq.mu.Lock()
 	defer fq.mu.Unlock()
 	fq.maxConcurrency = uint64(c.MaxConcurrency)
 	fq.maxQueueSize = uint64(c.MaxQueueSize)
 	fq.timeout = c.Timeout
+	fq.sem = semaphore.NewWeighted(int64(c.MaxConcurrency))
+	fq.counter.Store(uint64(0))
```

Comment:
I think this is not correct - there might be in-flight requests that would call Dec() resulting in overflow.

Reply:
I will check what happens in this case and make sure it will be 0

Reply:
I found another bug with the counter while reconfigure we have a race condition in accessing the counter which can show up in high routing table update rate (test with 5ms flapping fifo config).